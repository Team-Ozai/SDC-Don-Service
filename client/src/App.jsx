import React from 'react'
import axios from 'axios'
import styled from 'styled-components'
import Story from './Story.jsx'
import RisksAndChallenges from './RisksAndChallenges.jsx'
import EnvironmentalCommitments from './EnvironmentalCommitments.jsx'
import Nav from './style/Nav.js'
import NavContainer from './style/NavContainer.js'
import NavTitle from './style/NavTitle.js'
import Wrapper from './style/Wrapper.js'
import LeftMain from './style/LeftMain.js'
import Link from './style/Link.js'
import RefDiv from './style/RefDiv.js'

class Campaign extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      story : {},
      risksAndChallenges : {},
      environmentalCommitments : {},
    }

    this.storyText = React.createRef()
    this.risksAndChallengesText = React.createRef()
    this.environmentalCommitmentsText = React.createRef()

    this.refStory = React.createRef()
    this.refRisksAndChallenges = React.createRef()
    this.refEnvironmentalCommitments = React.createRef()

    this.handleClickStory = this.handleClickStory.bind(this)
    this.handleClickRisksAndChallenges = this.handleClickRisksAndChallenges.bind(this)
    this.handleClickEnvironmentalCommitments = this.handleClickEnvironmentalCommitments.bind(this)
    this.handleScroll = this.handleScroll.bind(this)
  }

  componentDidMount(){
    this.storyText.current.focus();
    window.addEventListener('scroll', this.handleScroll);
    this.fetchStory();
    this.fetchRisksAndChallenges();
    this.fetchEnvironmentalChallenges();
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.state.refStory);
  }

  focus(text) {
    if (text === "story") {
      this.storyText.current.focus();
    } else if (text === "risksAndChallenges") {
      this.risksAndChallengesText.current.focus()
    }else if (text === "environmentalCommitments") {
      this.environmentalCommitmentsText.current.focus()
    }
  }

  handleScroll(event) {
    if (
      window.pageYOffset >= (this.refStory.current.offsetTop-28)
      && window.pageYOffset <= (this.refStory.current.offsetTop + this.refStory.current.clientHeight-28)
      )
    {
      this.focus('story');
    }else if (
      window.pageYOffset >= (this.refRisksAndChallenges.current.offsetTop-28)
      && window.pageYOffset <= (this.refRisksAndChallenges.current.offsetTop
      + this.refRisksAndChallenges.current.clientHeight-28)
      )
    {
      this.focus('risksAndChallenges')
    }else if(
      window.pageYOffset >= (this.refEnvironmentalCommitments.current.offsetTop-28)
      && window.pageYOffset <= (this.refEnvironmentalCommitments.current.offsetTop
      + this.refEnvironmentalCommitments.current.clientHeight-28)
      )
    {
      this.focus('environmentalCommitments')
    }
  }

  scrollToRef(ref){
    window.scrollTo({
      top: ref.current.offsetTop-28
    });
  }

  fetchStory(){
    axios({
      method: 'get',
      url: '/api/story/50',
    })
    .then((response)=>{
      this.setState({
      story: response.data
      })
    })
    .catch((err)=>{
      console.log(err)
    })
  }

  fetchRisksAndChallenges(){
    axios({
      method: 'get',
      url: '/api/RisksAndChallenges/2',
    })
    .then((response)=>{
      this.setState({
        risksAndChallenges: response.data
      })
    })
    .catch((err)=>{
      console.log(err)
    })
  }

  fetchEnvironmentalChallenges(){
    axios({
      method: 'get',
      url: '/api/EnvironmentalCommitments/1',
    })
    .then((response)=>{
      this.setState({
        environmentalCommitments: response.data
      })
    })
    .catch((err)=>{
      console.log(err)
    })
  }

  handleClickStory(e){
    e.preventDefault();
    this.scrollToRef(this.refStory);
  }

  handleClickRisksAndChallenges(e){
    e.preventDefault();
    this.scrollToRef(this.refRisksAndChallenges);
  }

  handleClickEnvironmentalCommitments(e){
    e.preventDefault();
    this.scrollToRef(this.refEnvironmentalCommitments);
  }

  render (){
    return (
      <Wrapper>
        <NavContainer>
          <Nav>
              <NavTitle><Link href="#" onClick={this.handleClickStory} ref={this.storyText}>Story</Link></NavTitle>
              <NavTitle><Link href="#" onClick={this.handleClickRisksAndChallenges} ref ={this.risksAndChallengesText}>Risks And Challenges</Link></NavTitle>
              <NavTitle><Link href="#" onClick={this.handleClickEnvironmentalCommitments} ref = {this.environmentalCommitmentsText} >Environmental Commitments</Link></NavTitle>
          </Nav>
        </NavContainer>
        <LeftMain>
          <RefDiv ref={this.refStory}>
          <Story data = {this.state.story}/>
          </RefDiv>
          <RefDiv div ref={this.refRisksAndChallenges}>
          <RisksAndChallenges data = {this.state.risksAndChallenges}/>
          </RefDiv>
          <div ref={this.refEnvironmentalCommitments}>
          <EnvironmentalCommitments data = {this.state.environmentalCommitments} />
          </div>
        </LeftMain>
      </Wrapper>
    )
  }
}

export default Campaign