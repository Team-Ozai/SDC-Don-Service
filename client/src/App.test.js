import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import App from './App.jsx';
import NavTitle from './style/NavTitle.js'
import renderer from 'react-test-renderer'

Enzyme.configure({ adapter: new Adapter() });

describe('App Testing', ()=>{

  //smoke test
  it('renders without crashing', () => {
    shallow(<App />);
  });

  //snapshot test
  it('matches the snapshot', ()=>{
    const tree = renderer.create(<App/>).toJSON()
    expect(tree).toMatchSnapshot();
  })

  // it('should show text', ()=>{
  //   const wrapper = shallow(<App/>);
  //   const text = wrapper.find(NavTitle);
  //   expect (text.text()).toBe('Story2')
  // })
})