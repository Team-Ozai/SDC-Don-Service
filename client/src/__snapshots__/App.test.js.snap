// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing matches the snapshot 1`] = `
<div
  className="sc-AxheI eJXcSL"
>
  <div
    className="sc-AxhUy duQhtm"
  >
    <nav
      className="sc-AxhCb dkTehI"
    >
      <div
        className="sc-AxgMl kXAnQF"
      >
        <a
          className="sc-AxmLO indOiH"
          href="#"
          onClick={[Function]}
        >
          Story
        </a>
      </div>
      <div
        className="sc-AxgMl kXAnQF"
      >
        <a
          className="sc-AxmLO indOiH"
          href="#"
          onClick={[Function]}
        >
          Risks And Challenges
        </a>
      </div>
      <div
        className="sc-AxgMl kXAnQF"
      >
        <a
          className="sc-AxmLO indOiH"
          href="#"
          onClick={[Function]}
        >
          Environmental Commitments
        </a>
      </div>
    </nav>
  </div>
  <div
    className="sc-Axmtr hpCaXH"
  >
    <div
      className="sc-fzozJi eNfNpC"
    >
      <div>
        <h3
          className="sc-AxjAm gCQsTJ"
        >
          Story
        </h3>
        <p
          className="sc-AxirZ AEowp"
        />
        <div>
          <img
            className="sc-AxiKw hSLCUF"
          />
        </div>
        <p
          className="sc-AxirZ AEowp"
        />
      </div>
    </div>
    <div
      className="sc-fzozJi eNfNpC"
    >
      <div>
        <h3
          className="sc-AxjAm gCQsTJ"
        >
          Risks And Commitments
        </h3>
        <p
          className="sc-AxirZ AEowp"
        />
        <p
          className="sc-AxirZ AEowp"
        />
      </div>
    </div>
    <div>
      <div>
        <h3
          className="sc-AxjAm gCQsTJ"
        >
          Environmental Commitments
        </h3>
        <p
          className="sc-AxirZ AEowp"
        />
        <p
          className="sc-AxirZ AEowp"
        />
        <p
          className="sc-AxirZ AEowp"
        />
        <p
          className="sc-AxirZ AEowp"
        />
        <p
          className="sc-AxirZ AEowp"
        />
        <p
          className="sc-AxirZ AEowp"
        />
        <p
          className="sc-AxirZ AEowp"
        />
        <p
          className="sc-AxirZ AEowp"
        />
      </div>
    </div>
  </div>
</div>
`;
